R/get_predicted_args.R

```diff
@@ -0,0 +1,217 @@
+# process predict-specific arguments ------------------------------------------
+
+.get_predicted_args <- function(x,
+                                data = NULL,
+                                predict = "expectation",
+                                include_random = TRUE,
+                                include_smooth = TRUE,
+                                ci = 0.95,
+                                newdata = NULL,
+                                verbose = TRUE,
+                                ...) {
+
+  # check whether user possibly used the "type" instead of "predict" argument
+  dots <- list(...)
+
+  # one of "type" or "predict" must be provided...
+  if (is.null(dots$type) && is.null(predict)) {
+    stop(format_message("Please supply a value for the `predict` argument."))
+  }
+
+  # ...but not both
+  if (!is.null(dots$type) && !is.null(predict)) {
+    stop(format_message(
+      '`predict` and `type` cannot both be given. The preferred argument for `get_predicted()` is `predict`.',
+      'To use the `type` argument, set `predict = NULL` explicitly, e.g.,:',
+      '`get_predicted(model, predict = NULL, type = "response")`'
+    ))
+  }
+
+  # copy "type" to "predict"
+  if (!is.null(dots$type)) {
+    predict <- dots$type
+  }
+
+  if (length(predict) > 1) {
+    predict <- predict[1]
+    if (isTRUE(verbose)) {
+      msg <- format_message(sprintf("More than one option provided in `predict`. Using first option '%s' now."), predict[1])
+      warning(msg, call. = FALSE)
+    }
+  }
+
+  # Get info
+  info <- model_info(x, verbose = FALSE)
+
+  # Data
+  if (!is.null(newdata) && is.null(data)) data <- newdata
+  if (is.null(data)) data <- get_data(x, verbose = verbose)
+
+  # CI
+  if (is.null(ci)) ci <- 0
```

Comment:
Can we leave `ci` NULL and then skip all the confidence interval construction steps? It could be useful to get fast predictions without CI for bootstrapping, for example.

Reply:
good point